mesonbuild/build.py

```diff
@@ -1727,7 +1738,9 @@ def __post_init__(self) -> None:
         self.outfilelist: T.List[str] = []
         self.outmap: T.Dict[File, T.List[str]] = {}
         self.extra_depends = []  # XXX: Doesn't seem to be used?
-        self.depend_files: T.List[File] = []
+
+        if self.depend_files is None:
+            self.depend_files: T.List[File] = []
```

Comment:
With the `dataclasses.field` change you can drop all of this

Reply:
done. I've also corrected `self.extra_args` that is similar